sklearn/cluster/_birch.py

```diff
@@ -463,19 +462,21 @@ def fit(self, X, y=None):
         return self._fit(X)
 
     def _fit(self, X):
-        X = self._validate_data(X, accept_sparse='csr', copy=self.copy)
+        # partial_fit is called for the first time or fit is called
+        partial_fit = getattr(self, 'partial_fit_')
```

Comment:
Without seeing the code, I was a bit surprised that would work.
Would it be worth mentioning that `partial_fit_` is set when calling `fit()` or `partial_fit()`

Reply:
>  Would it be worth mentioning that partial_fit_ is set when calling fit() or partial_fit()

From what I can tell the `partial_fit_` attribute always exists. it's True when `partial_fit` was called, and False when `fit` was called.

Also we should just use `self.partial_fit_` instead of relying on getattr since we  don't specify a default.